library/std/src/io/stdio.rs

```diff
@@ -873,6 +875,49 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Reads a [`String`] from [standard input](Stdin). The trailing
+/// newline is stripped. Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_input)]
+/// use std::io;
+///
+/// fn main() -> io::Result<()> {
+///     print!("Enter name: ");
+///
+///     let name: String = io::read_line()?;
+///
+///     println!("Your name is {}!", name);
+///
+///     Ok(())
+/// }
+/// ```
+#[unstable(
+    feature = "io_input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn read_line() -> Result<String> {
+    stdout().flush()?; // print!() does not flush :(
+    let mut input = String::new();
+    match stdin().read_line(&mut input)? {
+        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly")),
+        _ => {
+            input.pop();
```

Comment:
Considering `stdin().read_line()` does *not* drop newlines, shouldn't this function keep those in tact as well? Might be a bit confusing that `stdin().read_line()` gives a line including the newline, and `io::read_line()` gives one without the newline.

I agree it's probably more useful if it already strips the newline. But I'm worried about the inconsistency between two functions with the same name.

Reply:
I agree the inconsistency between `stdin().read_line()` and `io::read_line()` is concerning, but I think wanting to keep the newline would be in the minority and could be corrected with this snippet of code, perhaps we can avoid confusion by changing the name of the function?

```rust
if cfg!(windows) {
    string.push_str("\r\n");
} else if cfg!(unix) {
    string.push('\n');
}
```

Reply:
> could be corrected with this snippet of code

No, that unconditionally adds a `\r\n` or `\n`. The original line might or might not end in a newline (e.g. at the end of the stream/file), and might be just a `\n` on Windows in certain circumstances. This is why the already existing `read_line` function keeps the newlines, to not throw away any information.

Anyway, I agree that a function that gives a line with the newline stripped is useful. But considering the existing `read_line` keeps the newlines, this shouldn't have different behaviour while using the same name. (Also, the existing `read_line` returns `Ok` for zero bytes of input, while this function returns an `Err` in that case.)

Is there an RFC for this? It's a bit hard right now to keep track of all the design decisions like the name, newline stripping, the possibility of parsing things, showing a prompt or not, etc.

Reply:
That seems like a good plan, i believe it was mentioned in the previous PR that there have been RFC's on this before.

Reply:
A better choice may be to change the naming to reflect that it does not include the new lines, at least that could solve the problem with naming inconsistency.

But if we keep the line, then people would need to fallback to `trim` which currently does not return a `String` in place but `&str`.